tools/pythonpkg/src/native/python_objects.cpp

```diff
@@ -575,7 +575,14 @@ py::object PythonObject::FromValue(const Value &val, const LogicalType &type,
 		auto &array_values = ArrayValue::GetChildren(val);
 		auto array_size = ArrayType::GetSize(type);
 		auto &child_type = ArrayType::GetChildType(type);
-		py::tuple arr(array_size);
+
+#ifdef __EMSCRIPTEN__
+		// without this cast, a static_assert fails in pybind11
+		auto tuple_size = static_cast<ssize_t>(array_size);
```

Comment:
Why is this failing a static_assert in pybind11 in the pyodide build but not outside of it?
The only thing we're changing here is the implementation of the CPython library right? Then I would expect the Pyodide to not have issues in pybind11 but in CPython internals?

Reply:
The issue is that `GetSize` returns an `idx_t`, which desugars to an `unsigned long long` when building with emscripten, and `py::tuple` has a `static_assert` that enforces that `sizeof(unsigned long long) <= sizeof(ssize_t)` which becomes `8 <= 4` which fails the assertion. Integer inputs to the `py::tuple` constructor are all cast to `ssize_t`.

It looks like `ssize_t` is 4 bytes when compiling with Emscripten, and `idx_t` is typedef'd to `uint64_t`.

Reply:
> The only thing we're changing here is the implementation of the CPython library right?

Not quite 😅 

The entire compilation toolchain is different, as well as the runtime.

Reply:
Should we not have the static_cast always on then, @Tishj ?

Basically we are passing an `idx_t` to function taking a `ssize_t`, so it should be fine doing the explicit conversion always (instead of the implicit one), that removes the ifdef and might make it more easier to maintain.

Reply:
I removed the `#ifdef`.